### ISSUE TITLE
Numeric parameter is serialized as string

### ISSUE THREAD

**Issue by igarashitm** [KaotoIO/kaoto#1653]
### Describe the Bug

Numeric parameter is serialized as string, observed on `onException` `maximumRedeliveries`

### Steps to Reproduce the Bug or Issue

1. Create `onException`
2. Set `maximumRedeliveries` to `3`
3. See source code, `maximumRedeliveries` is double quoted

### Screenshots or Videos

![Screenshot From 2024-11-21 15-59-17](https://github.com/user-attachments/assets/12927f02-deec-4962-ae7e-28eba103a23c)

![Screenshot From 2024-11-21 15-58-51](https://github.com/user-attachments/assets/0bfbe651-3b0c-4f30-ad58-e690584a182c)

### Platform

- OS: [e.g. macOS, Windows, Linux]
- Browser: [e.g. Chrome, Safari, Firefox]
- Version: [e.g. 0.30.0]

**Comment by apupier** [comment]
Shouldn't we be able to provide a String so that we can use placeholders? In this case the issue is in the yaml schema.

**Comment by igarashitm** [comment]
I don't think we want all the properties to be String in the schema itself. No matter how it's brought including with property placeholder, if at runtime it's finally a number, then schema should declare it's a number. The property placeholder is some magic beyond that I would say.

**Comment by lordrip** [comment]
Considering the purpose of the schema is to express to the user what can be written there, I believe the schema should allow for `string` types there. At the same time, considering that we can use placeholders almost everywhere, it's very likely all properties will be `string` :sweat_smile:.

I want to propose that when writing a number over a number field, make Kaoto understand it and write as-is, otherwise, write the string.

**Comment by lordrip** [comment]
For now we're going with serializing a number when is a number, in case of a placeholder, we serialize string.
